platform/platform_freebsd.cpp

```diff
@@ -0,0 +1,142 @@
+#include "platform/platform.hpp"
```

Comment:
@ygorshenin Virtual classes will make code much simpler and easier to support. And it’s very hard to customize member variables now for Platform (check Android implementation for example). I agree that we don’t need runtime polymorphism, but it’s just lesser evil.
And idea with defines to avoid unnecessary virtual calls sounds interesting :)

> On Oct 6, 2015, at 01:15, ygorshenin <notifications@github.com> wrote:
> 
> In platform/platform_freebsd.cpp <https://github.com/mapsme/omim/pull/112#discussion_r41235779>:
> 
> > @@ -0,0 +1,142 @@
> > +#include "platform/platform.hpp"
> @deathbaba <https://github.com/deathbaba> , why do we need virtuals at all? There is always one and only one implementation of Platform on each platform. It doesn't make sense to compile PlatformWin on Android, for example. We have been used separate compilation for Platform instead of virtual inheritance and it works fine. From my point of view virtuals are not better that current approach, because you'll get a bunch of PlatformWin, PlatformLinux, PlatformPosix, PlatformUnix, ... classes instead of a bunch of source files.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/112/files#r41235779>.
>